recipes/fast-dds/2.3.X/conanfile.py

```diff
@@ -145,7 +145,10 @@ def validate(self):
                 # This combination leads to an fast-dds error when linking
                 # linking dynamic '*.dll' and static MT runtime
                 raise ConanInvalidConfiguration("Mixing a dll {} library with a static runtime is a bad idea".format(self.name))
-
+        elif self.settings.os == "Macos":
+                # FIXME: Testing currently fails with MacOS - see https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/8561/1-configs/macos-clang/fast-dds/2.3.2//summary.json
+                # dyld: Library not loaded: libfastrtps.2.3.dylib --> Reason: image not found
+                raise ConanInvalidConfiguration("Macos currently not supported")
```

Comment:
I'm not aware of any change on the Macos machines, this should work as it was working before 🤔

Reply:
Also extremely strange to me - but have no Mac - so cannot check reason ....

Reply:
I've just tried locally and it worked for me (apple-clang 13). I'll investigate it in a different PR here in CCI

Reply:
Here is the link: https://github.com/conan-io/conan-center-index/pull/8566. I'm also upgrading dependencies, to apply some meaningful change.

Reply:
@jgsogo thanks for supporting here :)

Reply:
No but your test package is likely broken, you don't inject DYLD_LIBRARY_PATH by not relying on self.run with environment=True, so yes shared on Macos breaks obviously.

Reply:
1st thanks for pointing it out - yes -.- i forgot i start them with popen now -.- will adapt tomorrow. 
2nd The reason for this huge test is a bug hard to detect but happened when one dependency (foonathan memory) was set with wrong default options. then sending messages from one thread to one other inside same process works fine but the communication between 2 process crash on runtime. Basically i want to check if 1 message can be sent by one process and received by 1 process. i dont want accidentally again a package with this misbehavior.